go-controller/pkg/node/ovn_db_mgmt.go

```diff
@@ -0,0 +1,139 @@
+package node
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func (n *OvnNode) ensureOvnRaft(db string) {
+	ticker := time.NewTicker(60 * time.Second)
+	klog.Infof("Starting ensure routine for Raft db: %s", db)
+	kclient := n.Kube.(*kube.Kube)
+	for {
+		select {
+		case <-ticker.C:
+			if dbOnNode, _ := metrics.CheckPodRunsOnGivenNode(kclient.KClient.(*kubernetes.Clientset),
+				"name=ovnkube-db", n.name, false); dbOnNode {
+				klog.Infof("Node %s has a OVN DB pod, ensuring OVN Raft membership", n.name)
+				n.ensureLocalRaftServerID(db)
+				n.ensureClusterRaftMembership(db)
+			}
+		case <-n.stopChan:
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+// ensureLocalRaftServerID is used to ensure there is no stale member in the Raft cluster with our address
+func (n *OvnNode) ensureLocalRaftServerID(db string) {
+	var dbName string
+	var appCtl func(args ...string) (string, string, error)
+
+	if strings.Contains(db, "ovnnb") {
+		dbName = "OVN_Northbound"
+		appCtl = util.RunOVNNBAppCtl
+	} else {
+		dbName = "OVN_Southbound"
+		appCtl = util.RunOVNSBAppCtl
+	}
+
+	out, stderr, err := util.RunOVSDBTool("db-sid", db)
```

Comment:
considering running "ovs-appctl -t /var/run/ovn/ovnnb_db.ctl cluster/sid OVN_Northbound" to get the current server id? Getting from running state is probably better?

Reply:
i guess this comes down to what is a better source of truth, the database or the memory state? I would argue the database should always be the source of truth.